src/Features/Core/Portable/UseCollectionInitializer/AbstractUseCollectionInitializerDiagnosticAnalyzer.cs

```diff
@@ -145,257 +156,64 @@ private void AnalyzeNode(SyntaxNodeAnalysisContext context, INamedTypeSymbol ien
         }
 
         protected abstract ISyntaxFactsService GetSyntaxFactsService();
-    }
-
-    internal struct Analyzer<
-            TExpressionSyntax,
-            TStatementSyntax,
-            TObjectCreationExpressionSyntax, 
-            TMemberAccessExpressionSyntax,
-            TInvocationExpressionSyntax,
-            TExpressionStatementSyntax,
-            TVariableDeclaratorSyntax>
-        where TExpressionSyntax : SyntaxNode
-        where TStatementSyntax : SyntaxNode
-        where TObjectCreationExpressionSyntax : TExpressionSyntax
-        where TMemberAccessExpressionSyntax : TExpressionSyntax
-        where TInvocationExpressionSyntax : TExpressionSyntax
-        where TExpressionStatementSyntax : TStatementSyntax
-        where TVariableDeclaratorSyntax : SyntaxNode
```

Comment:
@jmarolf This was the old approach where there was a helper type for doing the analysis.  To be able to use this in both the analyzer *and* fixer, you had to refer, in both places, to the entire ```Analyzer<TExpressionSyntax, TStatementSyntax, TObjectCreationExpressionSyntax, TMemberAccessExpressionSyntax, TInvocationExpressionSyntax, TExpressionStatementSyntax, TVariableDeclaratorSyntax>``` type.  Very very annoying :)

The approach here allows us to just simply refer to types like Analyzer/Match/etc. without the need for all this type system finagling :)